### ISSUE TITLE
Add alt-texts to amusewiki

### ISSUE THREAD

**Issue by anarchistickaknihovna** [melmothx/amusewiki#421]
Currently the HTML alt attribute of images in amuse wiki articles
is the same as the filename of the image. Thus the alt text is
often gibberish, which leads to poor experience for users using screen
readers. (We tested this on Arch Linux with Orca, but it would be better
to get feedback from someone with visual impairment.)

Suggestions:
- Add a way for contributors to add alt-texts to images. This could
be done in the Amuse wiki markup or by adding the alt text directly
to uploaded images as metadata (this seems like an inferior solution
though, since the same image can and should have different alt-texts in
different contexts).
- Add alt-text support for pdfs. This seems like a more difficult thing
to implement, as there seems to be no standard way to include alt texts
in LaTeX (but there [are ways to do it](https://tex.stackexchange.com/questions/75102/need-alt-text-for-images-in-pdf-for-screen-readers)).
- Add tools to encourage users to use alt text. Example: warn
the contributor before publishing a text, if they forgot to provide
alt text to some images.
Many people don't know about the existence of alt texts and don't think about accessibility. We
think maintaining accessibility should be a community responsibility and the
software should steer people towards including it.

-The Anarchistická knihovna collective

**Comment by melmothx** [comment]
First and foremost: I don't think that amusewiki is doing so bad, accessibility wise. After all, we offer a wide range of formats for every taste.

Second: The alt text is something which is often important as a placeholder for broken images. The muse markup offers the possibility to set the caption for the image. The caption is visible everywhere, from PDF to EPUB to the web. If you think it's important that visually impaired people understand what's in the image, you should add a title. There's going to be some redundancy? Maybe, but it solves the matter in a practical and effective way.

Third: Adding a third pair of brackets for alt text in the markup *is* impractical, because it complicates code and usage. Honestly I still have problems, after all these years, to be sure, top of my head, if it comes first the filename or the description. Go figure with 3 pairs. Was it image, desc and alt? or image alt and desc? or something else entirely?

All that said, commit 57681dad adds the ability to add the alt_text in the /attachments/list console. The title will be used (if set) if the alt_text is missing. So I think this should close this issue, at least for the foreseeable future.
